plugins/jobs/girder_jobs/models/job.py

```diff
@@ -397,7 +399,12 @@ def updateJob(self, job, log=None, overwrite=False, status=None,
             job['updated'] = now
             updates['$set']['updated'] = now
 
-            updateResult = self.update(query, update=updates, multi=False)
+            try:
+                updateResult = self.update(query, update=updates, multi=False)
+            except DocumentTooLarge:
+                # It's plausible that the cause of this error is that the log has become too large
+                raise RestException('Document size exceeded', code=413)
```

Comment:
Typically models shouldn't throw RestExceptions. Let's handle this in the REST layer.

Reply:
See https://github.com/girder/girder/pull/3361#issue-678099222

Any `BaseModel.update()` could actually throw a DocumentTooLarge that I think should throw a 413.  Moving the RestException into BaseModel might be a better move.

I agree that models shouldn't really throw RestException, but since this is an error coming from pymongo and you'd have to catch it in hundreds of places, this seems like a reasonable exception to that rule.

What do you think?

Reply:
If you wanted to handle this more globally, I would suggest adding a `DocumentTooLarge` exception handler [here](https://github.com/girder/girder/blob/master/girder/api/rest.py#L682), and have that respond with 413. That keeps the model layer and HTTP layer separate, and if people want to handle `DocumentTooLarge` more ad hoc, they can still do so -- this would just act as a fallback for any unhandled ones in the REST layer.

Reply:
I'll do that, thanks.